src/workerd/server/tests/python/sdk/worker.py

```diff
@@ -508,6 +509,56 @@ def __init__(self, x):
     assert response_ws.status == 101
 
 
+async def response_buffer_source_unit_tests(env):
+    def make_array_buffer(values):
+        return js.Uint8Array.new(to_js(values)).buffer
+
+    buffer_source_factories = [
+        ("ArrayBuffer", lambda: make_array_buffer([1, 2, 3, 4, 5, 6, 7, 8])),
+        ("DataView", lambda: js.DataView.new(make_array_buffer([9, 10, 11, 12]))),
+    ]
+
+    typed_array_cases = [
+        ("Uint8Array", js.Uint8Array, [1, 2, 3, 4]),
+        ("Uint8ClampedArray", js.Uint8ClampedArray, [1, 2, 3, 4]),
+        ("Int8Array", js.Int8Array, [1, -1, 2, -2]),
+        ("Uint16Array", js.Uint16Array, [1, 2, 3, 4]),
+        ("Int16Array", js.Int16Array, [1, -2, 3, -4]),
+        ("Uint32Array", js.Uint32Array, [1, 2, 3, 4]),
+        ("Int32Array", js.Int32Array, [1, -2, 3, -4]),
+        ("Float32Array", js.Float32Array, [1.5, -2.5, 3.25, -4.75]),
+        ("Float64Array", js.Float64Array, [1.5, -2.5]),
+        ("BigInt64Array", js.BigInt64Array, [2**53 + 1, -(2**53 + 1), 2**54 + 2, -(2**54 + 2)]),
+        ("BigUint64Array", js.BigUint64Array, [2**53 + 1, 2**54 + 2, 2**55 + 3, 2**56 + 4]),
+        # TODO: Support Float16Array in Pyodide
+    ]
+
+    for type_name, constructor, values in typed_array_cases:
+        buffer_source_factories.append(
+            (
+                type_name,
+                lambda constructor=constructor, values=values: constructor.new(
+                    to_js(values)
+                ),
+            )
+        )
```

Comment:
Could we simplify and avoid having this loop by writing out the constructor calls directly?

```
buffers = [
  ("ArrayBuffer", make_array_buffer([1, 2, 3, 4, 5, 6, 7, 8])),
  ("DataView", js.DataView.new(make_array_buffer([9, 10, 11, 12]))),
  ("Uint8Array", js.Uint8Array.new([1, 2, 3, 4])),
  ...
```